### PR TITLE
Add RFC for more Random methods.

### PR DESCRIPTION
**Pull request by jemc**
[Rendered](https://github.com/ponylang/rfcs/blob/feature/more-random-methods/text/0000-more-random-methods.md).

### CODE REVIEW

**Review by Praetonus — COMMENTED** [review]
I think we should also add `iN` methods for signed integers, for convenience.

Also, what do you think about using `swap_elements` instead of `swap` for the new `Array` method? I've been thinking about using `swap` for a general idiom on collections and other types with value semantics to swap two objects. It could be used to cheaply "reconstruct" an embedded field, for example. I'll write an RFC on that eventually.

File: text/0000-more-random-methods.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+This RFC proposes the addition of various methods to the `Random` trait of the `random` package. It also proposes an additional method to the `Array` class.
+
+# Motivation
+
+The purpose of these additional methods is to increase the functionality of the `Random` trait.
+
+# Detailed design
+
+The following methods will be added to the `Random` trait, similar to the existing `u8`, `u16`, `u32`, `u54`, and `u128` methods:
```
Comment (Praetonus) [inline text/0000-more-random-methods.md]: There's a typo here. `u54` should be `u64`.

**Comment by jemc** [comment]
I'll update this with @Praetonus' requested additional methods soon, before the vote.

**Comment by mfelsche** [comment]
Just a small question: 

why is the int method specified for a exclusive range ``[0, max)`` and not ``[0, max]``. With the former it is not possible to generate values within the whole int range, which might be a nice feature for some use cases. Or is there a trivial way to convert a ``[0, max)`` to ``[0, max]``, keeping the distribution properties of the exclusive range?

But maybe this should be discussed in a different rfc.

**Comment by jemc** [comment]
@mfelsche Note that if you need to use the whole int range, you can use one of the other methods that doesn't use the floating point arithmetic - they will be faster / more direct anyway.

That is, instead of using `rand.int[U32](U32.max_value())`, you can just use `rand.u32()`

**Comment by mfelsche** [comment]
It turns out i misunderstood the documentation of ``random.Random`` which was perfectly clear on that.
Using the ``uX()`` and ``iX()`` methods (to be added) is all that is needed.
thanks for the clarification!

**Comment by jemc** [comment]
@Praetonus - I added the `iX` methods in ae5a959.

**Pull request merged**